spec/lib/areacodes_spec.rb

```diff
@@ -50,6 +50,15 @@
       end.sort_by {|r| -r.population }
       expect(cities).to eql(expected_records)
     end
+
+    it 'has only one state per area-code result' do
+      area_codes = described_class.details.map { |r| r.area_code }.uniq
+      area_codes.each do |ac|
+        major_cities = described_class.major_cities(ac)
+        states = major_cities.map{ |r| r.state_code }.uniq
+        expect(states.size).to eql(1), "Areacode: #{ac} has states: #{states}"
+      end
+    end
```

Comment:
Is the goal of this test to catch data errors?  There's no deduping logic in code right?

Reply:
That's right. There's no need for de-duping logic in the app code if it can be done during the generation of the data.